packages/amplify_core/lib/src/types/auth/session/auth_user.dart

```diff
@@ -3,36 +3,14 @@
 
 import 'package:amplify_core/amplify_core.dart';
 
-part 'auth_user.g.dart';
-
-@zAmplifyGenericSerializable
-class AuthUser<Details extends SignInDetails>
+abstract class AuthUser
```

Comment:
why to change to abstract?

Reply:
This is to match iOS/Android

Reply:
`AuthUser` is an interface in iOS and a final class in Android. I think it make sense to not change it to abstract because it is a data class with no behaviour.

Reply:
It is a template for a data class given that plugins would extend it and override the `SignInDetails` type. I'm okay making it instantiable in this case.